File: C/17087.c
#include <stdio.h>  // ¼û¹Ù²ÀÁú 6
#include <stdlib.h>

int gcd(int a, int b)
{
	if (b == 0)
		return a;
	else
		return gcd(b, a % b);
}


int main()
{
	int n, s;
	scanf("%d %d", &n, &s);

	int arr[100000];
	for (int i = 0; i < n; i++)
	{
		scanf("%d", &arr[i]);
		arr[i] = abs(arr[i] - s);
	}

	if (n == 1)
		printf("%d", arr[0]);
	else if (n == 2)
		printf("%d", gcd(arr[0], arr[1]));
	else
	{
		int result = gcd(arr[0], arr[1]);
		for (int i = 2; i < n; i++)
		{
			result = gcd(result, arr[i]);
		}
		printf("%d", result);
	}

	return 0;
}

File: JAVA/src/_17087.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

// 숨바꼭질 6
public class _17087 {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        String[] input = br.readLine().split(" ");

        int N = Integer.parseInt(input[0]);
        int S = Integer.parseInt(input[1]);
        String[] data = br.readLine().split(" ");

        int[] arr = new int[N];
        for (int i = 0; i < N; i++) {
            arr[i] = Math.abs(S - Integer.parseInt(data[i]));
        }

        int ans = arr[0];
        for (int i = 0; i < N; i++) {
            ans = gcd(ans, arr[i]);
        }

        System.out.println(ans);
    }

    public static int gcd(int a, int b) {
        int c;
        while (b != 0) {
            c = a % b;
            a = b;
            b = c;
        }

        return a;
    }
}
